backend/app/controllers/homes_controller.rb

```diff
@@ -5,7 +5,8 @@ class HomesController < ApplicationController
 
   # GET /homes or /homes.json
   def index
-    @homes = Home.all.first(30)
+    @homes = Home.where(search_params.slice(:street_number, :street_name, :unit_number, :city, :state, :zip_code,
+                                            :building_type, :assignment_id))
```

Comment:
Think you can kill the `.slice` here and also have this work.

```suggestion
    @homes = Home.where(search_params)
```